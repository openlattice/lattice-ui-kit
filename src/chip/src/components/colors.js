import * as Colors from '../../../colors';

const {
  NEUTRAL,
  PURPLE,
  BLUE,
  TEAL,
  GREEN,
  YELLOW,
  ORANGE,
  RED,
  MAGENTA,
  VIOLET
} = Colors;

export default {
  default: {
    background: NEUTRAL.N00,
    color: NEUTRAL.N600
  },
  blue: {
    background: BLUE.B00,
    color: BLUE.B400,
    clickable: BLUE.B100
  },
  green: {
    background: GREEN.G00,
    color: GREEN.G400,
    clickable: GREEN.G100
  },
  magenta: {
    background: MAGENTA.M00,
    color: MAGENTA.M400,
    clickable: MAGENTA.M100
  },
  orange: {
    background: ORANGE.O00,
    color: ORANGE.O500,
    clickable: ORANGE.O100
  },
  purple: {
    background: PURPLE.P00,
    color: PURPLE.P400,
    clickable: PURPLE.P100
  },
  red: {
    background: RED.R00,
    color: RED.R400,
    clickable: RED.R100
  },
  teal: {
    background: TEAL.T00,
    color: TEAL.T500,
    clickable: TEAL.T100
  },
  violet: {
    background: VIOLET.V00,
    color: VIOLET.V400,
    clickable: VIOLET.V100
  },
  yellow: {
    background: YELLOW.Y00,
    color: YELLOW.Y500,
    clickable: YELLOW.Y100
  },
  primary: {
    color: 'white'
  },
  secondary: {
    color: PURPLE.P400
  },
};
