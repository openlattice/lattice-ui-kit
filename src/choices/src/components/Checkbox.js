// @flow
import { forwardRef } from 'react';

import {
  CheckboxButtonIndicator,
  CheckboxChipIndicator,
  CheckboxIndicator,
  CheckboxInput,
  ChoiceInnerWrapper,
  ChoiceLabel,
  ChoiceText,
  ChoiceWrapper,
} from './styled';

import { stopPropagation } from '../../../utils';

type Props = {|
  checked ?:boolean;
  defaultChecked ?:boolean;
  disabled ?:boolean;
  forwardedRef ?:any;
  id ?:string;
  label ?:string;
  mode ?:string;
  name ?:string;
  onBlur ?:(event :SyntheticFocusEvent<HTMLInputElement>) => void;
  onChange ?:(event :SyntheticInputEvent<HTMLInputElement>) => void;
  onFocus ?:(event :SyntheticFocusEvent<HTMLInputElement>) => void;
  readOnly ?:boolean;
  value ?:any;
|};

/* eslint-disable react/jsx-props-no-spreading */
const Checkbox = ({
  disabled,
  forwardedRef,
  id,
  label,
  mode,
  readOnly,
  ...rest
} :Props) => (
  <ChoiceLabel htmlFor={id} readOnly={readOnly} disabled={disabled}>
    <ChoiceWrapper mode={mode}>
      <ChoiceInnerWrapper>
        <CheckboxInput
            id={id}
            readOnly={readOnly}
            disabled={disabled || readOnly}
            {...rest}
            ref={forwardedRef} />
        {
          mode === 'button' && label
            && <CheckboxButtonIndicator onClick={stopPropagation}>{label}</CheckboxButtonIndicator>
        }
        {
          mode === 'chip' && label
            && <CheckboxChipIndicator onClick={stopPropagation}>{label}</CheckboxChipIndicator>
        }
        { mode !== 'button' && mode !== 'chip' && <CheckboxIndicator />}
      </ChoiceInnerWrapper>
    </ChoiceWrapper>
    {
      (mode !== 'button' && mode !== 'chip') && (
        <ChoiceText>
          {label}
        </ChoiceText>
      )
    }
  </ChoiceLabel>
);
/* eslint-enable */

Checkbox.defaultProps = {
  checked: undefined,
  defaultChecked: undefined,
  disabled: false,
  forwardedRef: undefined,
  id: undefined,
  label: undefined,
  mode: undefined,
  name: undefined,
  onBlur: undefined,
  onChange: undefined,
  onFocus: undefined,
  readOnly: undefined,
  value: undefined,
};

export default forwardRef<Props, HTMLInputElement>((props, ref) => (
  /* eslint-disable react/jsx-props-no-spreading */
  <Checkbox {...props} forwardedRef={ref} />
  /* eslint-enable */
));
