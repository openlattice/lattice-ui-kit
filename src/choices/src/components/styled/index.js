// @ flow
import CheckboxIndicator from './CheckboxIndicator';
import ChoiceLabel from './ChoiceLabel';
import RadioIndicator from './RadioIndicator';
import { CheckboxInput, RadioInput } from './ChoiceInputs';

export {
  CheckboxIndicator,
  CheckboxInput,
  ChoiceLabel,
  RadioIndicator,
  RadioInput
};
