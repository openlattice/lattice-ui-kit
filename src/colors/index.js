/*
 * @flow
 */

/*
 * Inspiration:
 * https://atlassian.design/guidelines/brand/color-1
 * https://bitbucket.org/atlassian/atlaskit-mk-2/src/master/packages/core/theme
 * https://evergreen.surge.sh/components/colors
 */

import {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
} from './src/Colors';

export {
  BLACK,
  BLUE,
  BLUE_1,
  BLUE_2,
  GREEN,
  GREENS,
  GREEN_1,
  GREEN_2,
  GREEN_3,
  MAGENTA,
  MUSTARD_1,
  NEUTRAL,
  NEUTRALS,
  ORANGE,
  ORANGE_1,
  OVERLAY_BG,
  PEACH_1,
  PURPLE,
  PURPLES,
  RED,
  REDS,
  RED_1,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
  YELLOW_1,
};

export default {
  BLUE,
  GREEN,
  MAGENTA,
  NEUTRAL,
  ORANGE,
  PURPLE,
  RED,
  TEAL,
  VIOLET,
  WHITE,
  YELLOW,
};
