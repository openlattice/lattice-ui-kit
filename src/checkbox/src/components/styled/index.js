// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES } from '../../../../colors';

const CheckboxLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  font-size: 14px;
  font-weight: normal;
  margin: 10px 0;
  min-height: 20px;
  padding: 0 10px 0 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;


type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    position: absolute;
    left: 8px;
    top: 4px;
    width: 3px;
    height: 8px;
    border: solid white;
    border-width: 0 2px 2px 0;
    transform: rotate(45deg);
  `;
};

const CheckboxIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 3px;

  :after {
    ${getAfterStyles}
    }
  }
`;

const CheckboxInput = styled.input.attrs(() => ({
  type: 'checkbox'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;
  
  :hover + ${CheckboxIndicator},
  :focus + ${CheckboxIndicator} {
    background: ${NEUTRALS[1]};
  }

  :checked + ${CheckboxIndicator} {
    background: ${PURPLES[2]};
  }

  :checked:hover + ${CheckboxIndicator},
  :checked:focus + ${CheckboxIndicator} {
    background: ${PURPLES[1]};
  }

  :disabled + ${CheckboxIndicator} {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + ${CheckboxIndicator} {
    background: ${NEUTRALS[2]};
  }
`;


export {
  CheckboxIndicator,
  CheckboxInput,
  CheckboxLabel,
};
