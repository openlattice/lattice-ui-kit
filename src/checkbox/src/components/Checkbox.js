// @flow
import React, { Component } from 'react';
import { CheckboxIndicator, CheckboxInput, CheckboxLabel } from './styled';

type Props = {
  disabled ? :boolean;
  onChange ? :(event :SyntheticInputEvent<HTMLInputElement>) => mixed;
  value ? :any;
  label ? :string;
  name ? :string;
  checked ? :boolean;
  defaultChecked ? :boolean;
  readOnly ? :boolean;
}

type State = {
  isChecked :boolean;
};

class Checkbox extends Component<Props, State> {

  static defaultProps = {
    checked: undefined,
    defaultChecked: false,
    disabled: false,
    label: undefined,
    name: undefined,
    onChange: undefined,
    readOnly: undefined,
    value: undefined,
  };

  constructor(props :Props) {
    super(props);

    const { checked, defaultChecked } = props;
    this.state = {
      isChecked: (checked !== undefined)
        ? checked
        // $FlowFixMe optional prop provided by defaultProps
        : defaultChecked,
    };
  }

  // Use prop 'checked' over state 'isChecked' if available
  getIsChecked = () => {
    const { checked } = this.props;
    const { isChecked } = this.state;
    return checked !== undefined ? checked : isChecked;
  }

  handleChange = (e :SyntheticInputEvent<HTMLInputElement>) => {
    const { disabled, onChange, readOnly } = this.props;
    e.persist();
    if (!(disabled || readOnly)) {
      this.setState({ isChecked: e.target.checked });
      if (onChange) {
        onChange(e);
      }
    }
  };

  render() {
    const {
      checked,
      defaultChecked,
      label,
      onChange,
      readOnly,
      ...rest
    } = this.props;
    const isChecked = this.getIsChecked();
    return (
      <CheckboxLabel readOnly={readOnly}>
        {label}
        <CheckboxInput
            checked={isChecked}
            onChange={this.handleChange}
            {...rest} />
        <CheckboxIndicator checked={isChecked} />
      </CheckboxLabel>
    );
  }
}

export default Checkbox;
