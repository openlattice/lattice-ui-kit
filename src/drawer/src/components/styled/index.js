import styled, { css } from 'styled-components';

const getPosition = (props) => {
  const { side } = props;
  switch (side) {
    case 'right':
      return css`
        right: 0;
      `;

    case 'left':
    default:
      return css`
        left: 0;
      `;
  }
};

const DrawerWrapper = styled.div`
  position: fixed;
  background-color: white;
  height: 100%;
  width: 300px;
  top: 0;
  z-index: 900;
  box-shadow: 0 0 30px 0 rgba(0, 0, 0, 0.2);

  ${getPosition};
`;

export {
  // eslint-disable-next-line import/prefer-default-export
  DrawerWrapper
};
