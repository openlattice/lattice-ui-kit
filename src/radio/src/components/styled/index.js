// @flow
import styled, { css } from 'styled-components';
import { NEUTRALS, PURPLES, WHITE } from '../../../../colors';

const RadioLabel = styled.label`
  align-items: center;
  color: ${NEUTRALS[0]};
  display: inline-flex;
  margin: 10px 0;
  min-height: 20px;
  padding-left: 30px;
  pointer-events: ${props => (props.readOnly ? 'none' : 'auto')};
  position: relative;
`;

type IndicatorProps = {
  checked :boolean;
};

const getAfterStyles = ({ checked } :IndicatorProps) => {
  if (!checked) {
    return css`
      display: none;
    `;
  }

  return css`
    content: '';
    background-color: ${WHITE};
    border-radius: 50%;
    height: 8px;
    left: 6px;
    position: absolute;
    top: 6px;
    width: 8px;
  `;
};

const RadioIndicator = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: ${PURPLES[6]};
  border-radius: 50%;

  :after {
    ${getAfterStyles}
  }
`;

const RadioInput = styled.input.attrs(() => ({
  type: 'radio'
}))`
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;

  :hover + div,
  :focus + div {
    background: ${NEUTRALS[1]};
  }

  :checked + div {
    background: ${PURPLES[2]};
  }

  :checked:hover + div,
  :checked:focus + div {
    background: ${PURPLES[1]};
  }

  :disabled + div {
    background: ${PURPLES[6]};
    cursor: not-allowed;
  }

  :checked:disabled + div {
    background: ${NEUTRALS[2]};
  }
`;

export {
  RadioLabel,
  RadioInput,
  RadioIndicator,
};
