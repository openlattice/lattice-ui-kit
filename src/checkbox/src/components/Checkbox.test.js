import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Checkbox from './Checkbox';
import { CheckboxIndicator, CheckboxInput } from './styled';

import { PURPLES, NEUTRALS } from '../../../colors';

describe('Checkbox', () => {

  test('render matches snapshot', () => {
    const wrapper = mount(<Checkbox />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should pass through any props to the input', () => {
    const nativeProps = {
      disabled: true,
      name: 'test',
      required: true,
      form: 'test-form',
    };

    const wrapper = mount(<Checkbox {...nativeProps} />)
      .find('input')
      .props();
    expect(wrapper).toEqual(expect.objectContaining(nativeProps));
  });

  test('indicator styles should change appropriately', () => {
    const wrapper = mount(<Checkbox defaultChecked />);
    const checkboxInput = wrapper.find(CheckboxInput);

    // focus/hover
    expect(checkboxInput).toHaveStyleRule('background', NEUTRALS[1], {
      modifier: `:hover + ${CheckboxIndicator}`
    });
    expect(checkboxInput).toHaveStyleRule('background', NEUTRALS[1], {
      modifier: `:focus + ${CheckboxIndicator}`
    });

    // checked
    expect(checkboxInput).toHaveStyleRule('background', PURPLES[2], {
      modifier: `:checked + ${CheckboxIndicator}`
    });

    // checked + focus/hover
    expect(checkboxInput).toHaveStyleRule('background', PURPLES[1], {
      modifier: `:checked:focus + ${CheckboxIndicator}`
    });
    expect(checkboxInput).toHaveStyleRule('background', PURPLES[1], {
      modifier: `:checked:hover + ${CheckboxIndicator}`
    });

    // disabled styles
    expect(checkboxInput).toHaveStyleRule('background', PURPLES[6], {
      modifier: `:disabled + ${CheckboxIndicator}`
    });
    expect(checkboxInput).toHaveStyleRule('background', NEUTRALS[2], {
      modifier: `:checked:disabled + ${CheckboxIndicator}`
    });

  });

  describe('Uncontrolled <Checkbox />', () => {

    test('should be unchecked by default', () => {
      const wrapper = mount(<Checkbox />);
      expect(wrapper.find('input').prop('checked')).toEqual(false);
    });

    test('should not use state isChecked when prop checked is true', () => {
      const wrapper = mount(<Checkbox checked={false} />);
      const input = wrapper.find('input');
      input.simulate('change', { target: { checked: true } });
      expect(wrapper.find('input').prop('checked')).toEqual(false);
    });

    test('should call handleChange once', () => {
      const wrapper = mount(<Checkbox />);
      const instance = wrapper.instance();
      const handleChangeSpy = jest.spyOn(instance, 'handleChange');

      instance.forceUpdate();
      wrapper.find('input').simulate('change');

      expect(handleChangeSpy).toHaveBeenCalledTimes(1);
    });

    test('should toggle isChecked', () => {
      const wrapper = mount(<Checkbox />);

      wrapper.find(CheckboxInput).simulate('change', { target: { checked: true } });
      expect(wrapper.state().isChecked).toEqual(true);
      expect(wrapper.find('input').prop('checked')).toEqual(true);

      wrapper.find(CheckboxInput).simulate('change', { target: { checked: false } });
      expect(wrapper.state().isChecked).toEqual(false);
      expect(wrapper.find('input').prop('checked')).toEqual(false);
    });

    test('should not toggle isChecked when disabled', () => {
      const wrapper = mount(<Checkbox disabled />);
      wrapper.find(CheckboxInput).simulate('change', { target: { checked: true } });
      expect(wrapper.state().isChecked).toEqual(false);
    });
  });

  describe('Stateless <Checkbox checked />', () => {
    test('should not use state isChecked when prop checked is false', () => {
      const wrapper = mount(<Checkbox checked={false} />);
      const input = wrapper.find('input');
      input.simulate('change', { target: { checked: true } });
      expect(wrapper.find('input').prop('checked')).toEqual(false);
    });

    test('should not use state isChecked when prop checked is true', () => {
      const wrapper = mount(<Checkbox checked={false} />);
      const input = wrapper.find('input');
      input.simulate('change', { target: { checked: true } });
      expect(wrapper.find('input').prop('checked')).toEqual(false);
    });
  });

  describe('Controlled <Checkbox checked onChange={...} />', () => {

    test('should call onChange once', () => {
      const mockOnChange = jest.fn();
      const wrapper = mount(<Checkbox checked={false} onChange={mockOnChange} />);

      wrapper.find(CheckboxInput).simulate('change', { target: { checked: true } });
      expect(mockOnChange).toHaveBeenCalledTimes(1);
    });

    test('should not call onChange when disabled', () => {
      const mockOnChange = jest.fn();
      const wrapper = mount(<Checkbox checked={false} onChange={mockOnChange} disabled />);

      wrapper.find(CheckboxInput).simulate('change');
      expect(mockOnChange).toHaveBeenCalledTimes(0);
    });

  });

});
