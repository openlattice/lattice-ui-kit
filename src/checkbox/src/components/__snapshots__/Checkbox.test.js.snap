// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render matches snapshot 1`] = `
.c0 {
  display: inline-block;
  color: #555e6f;
  font-size: 14px;
  font-weight: normal;
  margin: 10px 0;
  min-height: 20px;
  padding: 0 10px 0 30px;
  pointer-events: auto;
  position: relative;
  vertical-align: middle;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  height: 20px;
  width: 20px;
  background: #e6e6f7;
  border-radius: 3px;
}

.c3:after {
  display: none;
}

.c1 {
  position: absolute;
  z-index: -1;
  opacity: 0;
  vertical-align: middle;
}

.c1:hover + .c2,
.c1:focus + .c2 {
  background: #8e929b;
}

.c1:checked + .c2 {
  background: #8045ff;
}

.c1:checked:hover + .c2,
.c1:checked:focus + .c2 {
  background: #6124e2;
}

.c1:disabled + .c2 {
  background: #e6e6f7;
  cursor: not-allowed;
}

.c1:checked:disabled + .c2 {
  background: #b6bbc7;
}

<Checkbox
  defaultChecked={false}
  disabled={false}
>
  <styled__CheckboxLabel>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styled__CheckboxLabel-sc-22ovv7-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:inline-block;color:",
              "#555e6f",
              ";font-size:14px;font-weight:normal;margin:10px 0;min-height:20px;padding:0 10px 0 30px;pointer-events:",
              [Function],
              ";position:relative;vertical-align:middle;",
            ],
          },
          "displayName": "styled__CheckboxLabel",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styled__CheckboxLabel-sc-22ovv7-0",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <label
        className="c0"
      >
        <styled__CheckboxInput
          checked={false}
          disabled={false}
          onChange={[Function]}
        >
          <StyledComponent
            checked={false}
            disabled={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "type": "checkbox",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "styled__CheckboxInput-sc-22ovv7-2",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "position:absolute;z-index:-1;opacity:0;vertical-align:middle;:hover + ",
                    ".c2",
                    ",:focus + ",
                    ".c2",
                    "{background:",
                    "#8e929b",
                    ";}:checked + ",
                    ".c2",
                    "{background:",
                    "#8045ff",
                    ";}:checked:hover + ",
                    ".c2",
                    ",:checked:focus + ",
                    ".c2",
                    "{background:",
                    "#6124e2",
                    ";}:disabled + ",
                    ".c2",
                    "{background:",
                    "#e6e6f7",
                    ";cursor:not-allowed;}:checked:disabled + ",
                    ".c2",
                    "{background:",
                    "#b6bbc7",
                    ";}",
                  ],
                },
                "displayName": "styled__CheckboxInput",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styled__CheckboxInput-sc-22ovv7-2",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onChange={[Function]}
          >
            <input
              checked={false}
              className="c1"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
          </StyledComponent>
        </styled__CheckboxInput>
        <styled__CheckboxIndicator
          checked={false}
        >
          <StyledComponent
            checked={false}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "c2",
                  "isStatic": false,
                  "lastClassName": "c3",
                  "rules": Array [
                    "position:absolute;top:0;left:0;height:20px;width:20px;background:",
                    "#e6e6f7",
                    ";border-radius:3px;:after{",
                    [Function],
                    "}}",
                  ],
                },
                "displayName": "styled__CheckboxIndicator",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "c2",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              checked={false}
              className="c2 c3"
            />
          </StyledComponent>
        </styled__CheckboxIndicator>
      </label>
    </StyledComponent>
  </styled__CheckboxLabel>
</Checkbox>
`;
